pontoon/base/views.py

```diff
@@ -294,19 +305,33 @@ def get_translations_from_other_locales(request):
         }, status=400)
 
     entity = get_object_or_404(Entity, pk=entity)
-    locales = entity.resource.project.locales.exclude(code=locale)
+    locale = get_object_or_404(Locale, code=locale)
+    preferred_locales = request.user.profile.preferred_locales
```

Comment:
We'll need to make sure the code also runs for anonymous users - right now it breaks.

So we'll need to introduce some check ([example](https://github.com/mozilla/pontoon/commit/c4bacd58e74cdcb1cbfeb8b56fc03692398d87ad)) before executing the code related to preferred locales.